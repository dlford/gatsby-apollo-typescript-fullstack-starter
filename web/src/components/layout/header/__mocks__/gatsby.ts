export default {
  ...jest.requireActual('gatsby'),
  graphql: jest.fn(),
  useStaticQuery: jest.fn(() => ({
    data: {
      site: {
        siteMetadata: {
          title: 'Gatsby Typescript Starter Kitchen Sink',
        },
      },
    },
  })),
}
