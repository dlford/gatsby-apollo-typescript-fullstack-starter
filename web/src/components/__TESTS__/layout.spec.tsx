import * as Gatsby from 'gatsby'
import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Layout from '../layout'

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      title: 'A Title',
    },
  },
}))

const paths = [
  '/',
  '/gatsby-typescript-starter-kitchen-sink/',
  '/some-other-path/',
]

paths.map((path) => {
  let asFragment: any

  beforeEach(() => {
    window.history.pushState({}, 'title', path)
    asFragment = render(
      <Layout location={window.location}>
        <div>hello</div>
      </Layout>,
    ).asFragment
  })

  afterEach(cleanup)

  describe('Layout', () => {
    it(`matches snapshot ${path}`, () => {
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
