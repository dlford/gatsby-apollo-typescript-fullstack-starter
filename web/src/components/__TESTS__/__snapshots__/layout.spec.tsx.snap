// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot / 1`] = `
<DocumentFragment>
  <div
    class="css-iy18fc"
  >
    <header
      class="css-fxtdit"
    >
      <div
        class="css-b1fiqv"
      >
        <div
          class="css-15ypf3i"
        >
          <svg
            viewBox="0 0 338.415 33.528"
          >
            <path
              d="M110.306 9.7l58.052 15.042 96.138-17.05L232.547.009 150.085 0z"
              fill="#00aad4"
            />
            <path
              d="M338.415.009l-.004 33.52L232.547.01z"
              fill="#2c89a0"
            />
            <path
              d="M73.062 24.734L150.085 0l-88.09.018-33.259 10.748c14.13 4.73 29.393 9.305 44.326 13.968z"
              fill="#1fc3f1"
            />
            <path
              d="M.016.01L0 31.25 61.996.022z"
              fill="#5fbcd3"
            />
          </svg>
        </div>
      </div>
      <div
        class="css-1n7se3o"
      >
        <svg
          viewBox="0 0 135.467 135.467"
        >
          <defs
            id="prefix__defs2"
          >
            <lineargradient
              id="prefix__linearGradient2393"
            >
              <stop
                id="prefix__stop2391"
                offset="0"
                stop-color="#f60"
                stop-opacity="1"
              />
            </lineargradient>
            <clippath
              clipPathUnits="userSpaceOnUse"
              id="prefix__clipPath32"
            >
              <path
                d="M0 2559.99V0h2560v2559.99z"
                id="prefix__path30"
              />
            </clippath>
          </defs>
          <g
            id="prefix__layer1"
            transform="translate(30.89 -192.422)"
          >
            <circle
              cx="43.043"
              cy="254.405"
              fill="#fff"
              fill-opacity="1"
              fill-rule="nonzero"
              id="prefix__path2444"
              r="56.772"
              stroke="none"
              stroke-width="0.222"
            />
            <path
              d="M42.91 199.717c-30.132 0-54.555 24.423-54.555 54.554 0 30.132 24.423 54.555 54.554 54.555 30.132 0 54.555-24.423 54.555-54.555 0-30.131-24.423-54.554-54.555-54.554zM.077 254.839l42.264 42.264C19.132 296.802.379 278.048.078 254.839zm52.42 41.195L1.147 244.683c4.355-19.042 21.398-33.252 41.762-33.252 14.232 0 26.845 6.943 34.636 17.627l-5.931 5.233c-6.32-9.06-16.82-14.991-28.705-14.991-15.13 0-28.019 9.61-32.89 23.06l44.8 44.797c10.88-3.941 19.247-13.124 22.05-24.498H58.301v-8.388h27.452c-.004 20.364-14.215 37.407-33.256 41.763z"
              fill="#744c9e"
              id="prefix__path815"
              stroke-width="0.044"
            />
            <g
              clip-path="none"
              id="prefix__g24"
              stroke="#fffffc"
              stroke-dasharray="none"
              stroke-miterlimit="4"
              stroke-opacity="1"
              stroke-width="22.506"
              transform="matrix(.04328 0 0 -.04328 -12.494 309.674)"
            >
              <g
                id="prefix__g26"
                stroke="#fffffc"
                stroke-dasharray="none"
                stroke-miterlimit="4"
                stroke-opacity="1"
                stroke-width="22.506"
              >
                <g
                  clip-path="url(#prefix__clipPath32)"
                  id="prefix__g28"
                  stroke="#fffffc"
                  stroke-dasharray="none"
                  stroke-miterlimit="4"
                  stroke-opacity="1"
                  stroke-width="22.506"
                >
                  <path
                    d="M2513.39 1622.31c-9.82 35.36-46.34 56.13-81.85 46.27-35.37-9.82-56.1-46.46-46.28-81.83 27.73-99.85 41.78-203.06 41.78-306.76 0-632.478-514.56-1147.021-1147.04-1147.021S132.969 647.512 132.969 1279.99c0 632.49 514.551 1147.04 1147.031 1147.04 267.93 0 527.18-95.18 731.79-264.56-7.96-19-12.38-39.85-12.38-61.73 0-88.35 71.62-159.97 159.97-159.97s159.97 71.62 159.97 159.97-71.62 159.97-159.97 159.97c-17.01 0-33.38-2.71-48.76-7.63-230.4 196.35-525.49 306.91-830.62 306.91-705.789 0-1280-574.2-1280-1280C0 574.199 574.211 0 1280 0c705.79 0 1280 574.199 1280 1279.99 0 115.72-15.67 230.9-46.61 342.32zm-911.12-847.509h276.3L1437.5 1919.5h-305.49L690.941 774.801h276.321l72.078 193.898h416.75l-75.44 214.561h-274.4l178.5 492.54 317.52-900.999"
                    fill="#192a45"
                    fill-opacity="1"
                    fill-rule="nonzero"
                    id="prefix__path34"
                    stroke="#fffffc"
                    stroke-dasharray="none"
                    stroke-miterlimit="4"
                    stroke-opacity="1"
                    stroke-width="22.506"
                  />
                </g>
              </g>
            </g>
            <path
              d="M-27.424 277.88v4.554h14.53v41.355h10.31v-41.355h14.53v-4.47c0-2.515 0-4.555-.111-4.611 0-.084-8.858-.112-19.616-.112l-19.56.084v4.583zm65.301-4.667c2.85.67 5.03 1.956 6.986 3.996 1.034 1.118 2.57 3.074 2.682 3.577 0 .167-4.834 3.437-7.768 5.253-.112.084-.559-.391-1.006-1.118-1.453-2.068-2.934-2.962-5.253-3.13-3.353-.223-5.588 1.537-5.588 4.471 0 .894.167 1.397.503 2.124.754 1.537 2.151 2.459 6.482 4.359 7.992 3.437 11.457 5.7 13.552 8.941 2.375 3.633 2.906 9.333 1.314 13.608-1.789 4.667-6.148 7.824-12.379 8.858-1.956.335-6.427.28-8.522-.084-4.471-.838-8.746-3.073-11.373-5.951-1.034-1.118-3.018-4.108-2.906-4.304l1.062-.67 4.191-2.431 3.158-1.844.726.978c.922 1.453 2.99 3.409 4.192 4.08 3.632 1.871 8.494 1.62 10.897-.56 1.034-.95 1.481-1.956 1.481-3.353 0-1.285-.196-1.872-.838-2.85-.894-1.23-2.683-2.235-7.712-4.47-5.784-2.46-8.243-4.024-10.535-6.427-1.313-1.453-2.514-3.717-3.073-5.589-.42-1.62-.56-5.588-.168-7.181 1.202-5.588 5.421-9.5 11.457-10.618 1.955-.391 6.566-.224 8.494.28z"
              fill="#007acc"
              fill-opacity="1"
              id="prefix__path2462"
              stroke="#fffffc"
              stroke-dasharray="none"
              stroke-miterlimit="4"
              stroke-opacity="1"
              stroke-width="0.856"
            />
          </g>
          <style
            id="style842"
          />
          <style
            id="style2458"
          />
        </svg>
      </div>
      <h1
        class="css-o7noaf"
      >
        A Title
      </h1>
    </header>
    <main>
      <div>
        hello
      </div>
    </main>
    <div
      class="css-14lv0my"
    >
      <footer
        class="css-1c0clj9"
      >
        <p>
          Copyleft 
          <span
            class="css-1k0tmud"
          >
            Â©
          </span>
           DL Ford 2020
        </p>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Layout matches snapshot /gatsby-typescript-starter-kitchen-sink/ 1`] = `
<DocumentFragment>
  <div
    class="css-iy18fc"
  >
    <header
      class="css-fxtdit"
    >
      <div
        class="css-b1fiqv"
      >
        <div
          class="css-15ypf3i"
        >
          <svg
            viewBox="0 0 338.415 33.528"
          >
            <path
              d="M110.306 9.7l58.052 15.042 96.138-17.05L232.547.009 150.085 0z"
              fill="#00aad4"
            />
            <path
              d="M338.415.009l-.004 33.52L232.547.01z"
              fill="#2c89a0"
            />
            <path
              d="M73.062 24.734L150.085 0l-88.09.018-33.259 10.748c14.13 4.73 29.393 9.305 44.326 13.968z"
              fill="#1fc3f1"
            />
            <path
              d="M.016.01L0 31.25 61.996.022z"
              fill="#5fbcd3"
            />
          </svg>
        </div>
      </div>
      <div
        class="css-1n7se3o"
      >
        <svg
          viewBox="0 0 135.467 135.467"
        >
          <defs
            id="prefix__defs2"
          >
            <lineargradient
              id="prefix__linearGradient2393"
            >
              <stop
                id="prefix__stop2391"
                offset="0"
                stop-color="#f60"
                stop-opacity="1"
              />
            </lineargradient>
            <clippath
              clipPathUnits="userSpaceOnUse"
              id="prefix__clipPath32"
            >
              <path
                d="M0 2559.99V0h2560v2559.99z"
                id="prefix__path30"
              />
            </clippath>
          </defs>
          <g
            id="prefix__layer1"
            transform="translate(30.89 -192.422)"
          >
            <circle
              cx="43.043"
              cy="254.405"
              fill="#fff"
              fill-opacity="1"
              fill-rule="nonzero"
              id="prefix__path2444"
              r="56.772"
              stroke="none"
              stroke-width="0.222"
            />
            <path
              d="M42.91 199.717c-30.132 0-54.555 24.423-54.555 54.554 0 30.132 24.423 54.555 54.554 54.555 30.132 0 54.555-24.423 54.555-54.555 0-30.131-24.423-54.554-54.555-54.554zM.077 254.839l42.264 42.264C19.132 296.802.379 278.048.078 254.839zm52.42 41.195L1.147 244.683c4.355-19.042 21.398-33.252 41.762-33.252 14.232 0 26.845 6.943 34.636 17.627l-5.931 5.233c-6.32-9.06-16.82-14.991-28.705-14.991-15.13 0-28.019 9.61-32.89 23.06l44.8 44.797c10.88-3.941 19.247-13.124 22.05-24.498H58.301v-8.388h27.452c-.004 20.364-14.215 37.407-33.256 41.763z"
              fill="#744c9e"
              id="prefix__path815"
              stroke-width="0.044"
            />
            <g
              clip-path="none"
              id="prefix__g24"
              stroke="#fffffc"
              stroke-dasharray="none"
              stroke-miterlimit="4"
              stroke-opacity="1"
              stroke-width="22.506"
              transform="matrix(.04328 0 0 -.04328 -12.494 309.674)"
            >
              <g
                id="prefix__g26"
                stroke="#fffffc"
                stroke-dasharray="none"
                stroke-miterlimit="4"
                stroke-opacity="1"
                stroke-width="22.506"
              >
                <g
                  clip-path="url(#prefix__clipPath32)"
                  id="prefix__g28"
                  stroke="#fffffc"
                  stroke-dasharray="none"
                  stroke-miterlimit="4"
                  stroke-opacity="1"
                  stroke-width="22.506"
                >
                  <path
                    d="M2513.39 1622.31c-9.82 35.36-46.34 56.13-81.85 46.27-35.37-9.82-56.1-46.46-46.28-81.83 27.73-99.85 41.78-203.06 41.78-306.76 0-632.478-514.56-1147.021-1147.04-1147.021S132.969 647.512 132.969 1279.99c0 632.49 514.551 1147.04 1147.031 1147.04 267.93 0 527.18-95.18 731.79-264.56-7.96-19-12.38-39.85-12.38-61.73 0-88.35 71.62-159.97 159.97-159.97s159.97 71.62 159.97 159.97-71.62 159.97-159.97 159.97c-17.01 0-33.38-2.71-48.76-7.63-230.4 196.35-525.49 306.91-830.62 306.91-705.789 0-1280-574.2-1280-1280C0 574.199 574.211 0 1280 0c705.79 0 1280 574.199 1280 1279.99 0 115.72-15.67 230.9-46.61 342.32zm-911.12-847.509h276.3L1437.5 1919.5h-305.49L690.941 774.801h276.321l72.078 193.898h416.75l-75.44 214.561h-274.4l178.5 492.54 317.52-900.999"
                    fill="#192a45"
                    fill-opacity="1"
                    fill-rule="nonzero"
                    id="prefix__path34"
                    stroke="#fffffc"
                    stroke-dasharray="none"
                    stroke-miterlimit="4"
                    stroke-opacity="1"
                    stroke-width="22.506"
                  />
                </g>
              </g>
            </g>
            <path
              d="M-27.424 277.88v4.554h14.53v41.355h10.31v-41.355h14.53v-4.47c0-2.515 0-4.555-.111-4.611 0-.084-8.858-.112-19.616-.112l-19.56.084v4.583zm65.301-4.667c2.85.67 5.03 1.956 6.986 3.996 1.034 1.118 2.57 3.074 2.682 3.577 0 .167-4.834 3.437-7.768 5.253-.112.084-.559-.391-1.006-1.118-1.453-2.068-2.934-2.962-5.253-3.13-3.353-.223-5.588 1.537-5.588 4.471 0 .894.167 1.397.503 2.124.754 1.537 2.151 2.459 6.482 4.359 7.992 3.437 11.457 5.7 13.552 8.941 2.375 3.633 2.906 9.333 1.314 13.608-1.789 4.667-6.148 7.824-12.379 8.858-1.956.335-6.427.28-8.522-.084-4.471-.838-8.746-3.073-11.373-5.951-1.034-1.118-3.018-4.108-2.906-4.304l1.062-.67 4.191-2.431 3.158-1.844.726.978c.922 1.453 2.99 3.409 4.192 4.08 3.632 1.871 8.494 1.62 10.897-.56 1.034-.95 1.481-1.956 1.481-3.353 0-1.285-.196-1.872-.838-2.85-.894-1.23-2.683-2.235-7.712-4.47-5.784-2.46-8.243-4.024-10.535-6.427-1.313-1.453-2.514-3.717-3.073-5.589-.42-1.62-.56-5.588-.168-7.181 1.202-5.588 5.421-9.5 11.457-10.618 1.955-.391 6.566-.224 8.494.28z"
              fill="#007acc"
              fill-opacity="1"
              id="prefix__path2462"
              stroke="#fffffc"
              stroke-dasharray="none"
              stroke-miterlimit="4"
              stroke-opacity="1"
              stroke-width="0.856"
            />
          </g>
          <style
            id="style842"
          />
          <style
            id="style2458"
          />
        </svg>
      </div>
      <h1
        class="css-o7noaf"
      >
        A Title
      </h1>
    </header>
    <main>
      <div>
        hello
      </div>
    </main>
    <div
      class="css-14lv0my"
    >
      <footer
        class="css-1c0clj9"
      >
        <p>
          Copyleft 
          <span
            class="css-1k0tmud"
          >
            Â©
          </span>
           DL Ford 2020
        </p>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Layout matches snapshot /some-other-path/ 1`] = `
<DocumentFragment>
  <div
    class="css-iy18fc"
  >
    <header
      class="css-fxtdit"
    >
      <div
        class="css-1uz0ojb"
      >
        <div
          class="css-15ypf3i"
        >
          <svg
            viewBox="0 0 338.415 33.528"
          >
            <path
              d="M110.306 9.7l58.052 15.042 96.138-17.05L232.547.009 150.085 0z"
              fill="#00aad4"
            />
            <path
              d="M338.415.009l-.004 33.52L232.547.01z"
              fill="#2c89a0"
            />
            <path
              d="M73.062 24.734L150.085 0l-88.09.018-33.259 10.748c14.13 4.73 29.393 9.305 44.326 13.968z"
              fill="#1fc3f1"
            />
            <path
              d="M.016.01L0 31.25 61.996.022z"
              fill="#5fbcd3"
            />
          </svg>
        </div>
      </div>
      <h1
        class="css-o7noaf"
      >
        A Title
      </h1>
    </header>
    <main>
      <div>
        hello
      </div>
    </main>
    <div
      class="css-14lv0my"
    >
      <footer
        class="css-1c0clj9"
      >
        <p>
          Copyleft 
          <span
            class="css-1k0tmud"
          >
            Â©
          </span>
           DL Ford 2020
        </p>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
